### ISSUE TITLE
ConnectExceptionMapper javax.ws.rs.NotFoundException: HTTP 404 Not Found 

### ISSUE THREAD

**Issue by Ziauddin135** [confluentinc/kafka-connect-jdbc#855]
I am running jdbc source connector in distributed mode. while the tasks of the connectors shows running since 3 days the data put in oracle with proper timestamp is not being picked up by connector since last 3 days. and only after i restarted my worker it started picking records. could you please help if this is due to database connectivity issue and after worker restart connection established how can it be resolved and taken care if tasks in exposed REST API of connectors continue to show RUNNING. kindly help 

below is the stack trace :-

==========================================================
 ERROR Uncaught exception in REST call to /favicon.ico (org.apache.kafka.connect.runtime.rest.errors.ConnectExceptionMapper:61)
javax.ws.rs.NotFoundException: HTTP 404 Not Found
	at org.glassfish.jersey.server.ServerRuntime$1.run(ServerRuntime.java:250)
	at org.glassfish.jersey.internal.Errors$1.call(Errors.java:248)
	at org.glassfish.jersey.internal.Errors$1.call(Errors.java:244)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:292)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:274)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:244)
	at org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:265)
	at org.glassfish.jersey.server.ServerRuntime.process(ServerRuntime.java:232)
	at org.glassfish.jersey.server.ApplicationHandler.handle(ApplicationHandler.java:679)
	at org.glassfish.jersey.servlet.WebComponent.serviceImpl(WebComponent.java:392)
	at org.glassfish.jersey.servlet.WebComponent.service(WebComponent.java:346)
	at org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:365)
	at org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:318)
	at org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:205)
	at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:873)
	at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:542)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:255)
	at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:1700)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:255)
	at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1345)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:203)
	at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:480)
	at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:1667)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:201)
	at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1247)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:144)
	at org.eclipse.jetty.server.handler.ContextHandlerCollection.handle(ContextHandlerCollection.java:220)
	at org.eclipse.jetty.server.handler.StatisticsHandler.handle(StatisticsHandler.java:174)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132)
	at org.eclipse.jetty.server.Server.handle(Server.java:505)
	at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:370)
	at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:267)
	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:305)
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:103)
	at org.eclipse.jetty.io.ChannelEndPoint$2.run(ChannelEndPoint.java:117)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.runTask(EatWhatYouKill.java:333)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:310)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.tryProduce(EatWhatYouKill.java:168)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:126)
	at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:366)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:698)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:804)

**Comment by Ziauddin135** [comment]
connector frequently running to this error. and the only way we can get it working is
pause connector
restart tasks
resume connector

no idea why connector tasks goes in sleep frequently. has this anything to do with thread preemption or context switching? can you suggest why once went in sleep thread doesnot wake up at its own unless above 3 step REST API are called

**Comment by Ziauddin135** [comment]
hi @C0urante ,

This seems to be similar to bug 677  i am using 5.3.1 version and my connector seems go in sleep oftenly and gets stuck to load data from source oracle db. only when i pause connector==>then restart task==> then resume connector it goes in wake up mode. can you help?

sorry for addressing direclty could you please help

Regards,
Zia

**Comment by C0urante** [comment]
Hi @Ziauddin135,

Unfortunately I don't have a lot of time available to work on this issue right now. I can say that the stack trace you've provided in the description looks unrelated to #677, though. If you see any other errors you may want to post them but if the connectors and tasks appear in the `RUNNING` state it's not guaranteed that there should be any or that if there are any, that they're indicative of a root cause.

You might also want to provide the configuration you're using for the connector here as it might be useful to anyone who wants to help.

Cheers,

Chris

**Comment by Ziauddin135** [comment]
![image](https://user-images.githubusercontent.com/25635087/85321372-3dc2b300-b4e2-11ea-9c50-e601d0c35eff.png)

**Comment by Ziauddin135** [comment]
Below is the config

```
{
connector.class: "io.confluent.connect.jdbc.JdbcSourceConnector",
timestamp.column.name: "UPDT_TS",
transforms.createKey.type: "org.apache.kafka.connect.transforms.ValueToKey",
connection.password: "${file:/usr/local/kafka/kafka/secret.properties:oraPassword}",
tasks.max: "1",
query: "select sys_id||'-'||brand||'-'||product|| as "id", sys_id as "sys_id",sales_org as "sales_org",brand as "brand",product as "product",updt_ts FROM prod_avl",
connection.attempts: "10000",
transforms: "createKey,extractInt",
transforms.extractInt.type: "org.apache.kafka.connect.transforms.ExtractField$Key",
batch.max.rows: "20000",
table.types: "query",
mode: "timestamp",
topic.prefix: "avail_topic",
transforms.extractInt.field: "id",
schema.pattern: "",
transforms.createKey.fields: "id",
connection.user: "${file:/usr/local/kafka/kafka/secret.properties:oraUsername}",
poll.interval.ms: "10000",
name: "jdbc_connector",
connection.url: "jdbc:oracle:thin:@host:port:db"
}
```

**Comment by Lumotheninja** [comment]
Not so sure what is the cause but I have also run into this bug recently.

[2020-08-20 18:22:58,083] ERROR Uncaught exception in REST call to /favicon.ico (org.apache.kafka.connect.runtime.rest.errors.ConnectExceptionMapper:61)
javax.ws.rs.NotFoundException: HTTP 404 Not Found
        at org.glassfish.jersey.server.ServerRuntime$1.run(ServerRuntime.java:274)
        at org.glassfish.jersey.internal.Errors$1.call(Errors.java:272)
        at org.glassfish.jersey.internal.Errors$1.call(Errors.java:268)
        at org.glassfish.jersey.internal.Errors.process(Errors.java:316)
        at org.glassfish.jersey.internal.Errors.process(Errors.java:298)
        at org.glassfish.jersey.internal.Errors.process(Errors.java:268)
        at org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:289)
        at org.glassfish.jersey.server.ServerRuntime.process(ServerRuntime.java:256)
        at org.glassfish.jersey.server.ApplicationHandler.handle(ApplicationHandler.java:703)
        at org.glassfish.jersey.servlet.WebComponent.serviceImpl(WebComponent.java:416)
        at org.glassfish.jersey.servlet.WebComponent.service(WebComponent.java:370)
        at org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:389)
        at org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:342)
        at org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:229)
        at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:865)
        at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:535)
        at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:255)
        at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:1595)
        at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:255)
        at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1317)
        at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:203)
        at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:473)
        at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:1564)
        at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:201)
        at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1219)
        at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:144)
        at org.eclipse.jetty.server.handler.HandlerCollection.handle(HandlerCollection.java:126)
        at org.eclipse.jetty.server.handler.StatisticsHandler.handle(StatisticsHandler.java:169)
        at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132)
        at org.eclipse.jetty.server.Server.handle(Server.java:531)
        at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:352)
        at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:260)
        at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:281)
        at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:102)
        at org.eclipse.jetty.io.ChannelEndPoint$2.run(ChannelEndPoint.java:118)
        at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.runTask(EatWhatYouKill.java:333)
        at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:310)
        at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.tryProduce(EatWhatYouKill.java:168)
        at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.produce(EatWhatYouKill.java:132)
        at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:762)
        at org.eclipse.jetty.util.thread.QueuedThreadPool$2.run(QueuedThreadPool.java:680)
        at java.lang.Thread.run(Thread.java:748)
[2020-08-20 18:22:58,090] INFO 100.66.227.2 - - [20/Aug/2020:09:22:58 +0000] "GET /favicon.ico HTTP/1.1" 404 49  18 (org.apache.kafka.connect.runtime.rest.RestServer:60)

Will investigate a bit and post whatever I can find

**Comment by Lumotheninja** [comment]
@Ziauddin135, my issue was a misconfigured zookeeper node. It was not part of the zookeeper cluster but started with a different zookeeper config. Once we fixed that this issue is no longer there. Hope this helps
